app/scripts/lib/local-store.js

```diff
@@ -114,6 +115,12 @@ export default class ExtensionStore {
       });
     });
   }
+
+  cleanUpMostRecentRetrievedState() {
+    if (this.mostRecentRetrievedState) {
+      this.mostRecentRetrievedState = null;
```

Comment:
Note that this is still getting set anew each time we persist state, which happens very often (every 200 ms or so at most).

We'll need to prevent it from being saved again as well. Maybe we could set an internal flag to say whether the wallet has been initialized.

Reply:
Thanks for suggestion , this is captured in [62780](https://github.com/MetaMask/metamask-extension/pull/26206/commits/62780be327ae1b0cb4590edeee7c07b7e738f7f8)